system/jlib/jtrace.cpp

```diff
@@ -202,8 +202,8 @@ class JLogSpanExporter final : public opentelemetry::sdk::trace::SpanExporter
                 out.setf("{ \"name\": \"%s\"", span->GetName().data());
                 out.append(", \"trace_id\": \"").append(32, traceID).append("\"");
                 out.append(", \"span_id\": \"").append(16, spanID).append("\"");
-                out.appendf(", \"start\": %ld", std::chrono::duration_cast<std::chrono::nanoseconds>(span->GetStartTime().time_since_epoch()).count());
-                out.appendf(", \"duration\": %ld", std::chrono::duration_cast<std::chrono::nanoseconds>(span->GetDuration()).count());
+                out.appendf(", \"start\": %s", std::to_string(span->GetStartTime().time_since_epoch().count()).c_str());
```

Comment:
A more standard approach would be to cast the value to a known integer type:

`out.appendf(", \"start\": %lld", (__int64)std::chrono::duration_cast<std::chrono::nanoseconds>(span->GetStartTime().time_since_epoch()).count());
`

Reply:
Strictly speaking you should have the the conversion to nanoseconds.  It just so happens the default Linux/gcc tick frequency is per nanosecond, but there is no guarantee.

Reply:
sdk::trace::SpanData.GetStartTime() returns opentelemetry::common::SystemTimestamp

SystemTimestamp.time_since_epocho return chrono::nanoseconds 

**std::chrono::nanoseconds** time_since_epoch() const noexcept  {    return td::**chrono::nanoseconds{nanos_since_epoch_**};  

I don't see the need to cast the retrn to chrono::nano, however we do apparently need to cast the result from count to long long for portability